src/antsibull/build_ansible_commands.py

```diff
@@ -397,6 +397,84 @@ def _extract_python_requires(
     )
 
 
+# pylint: disable-next=too-many-branches
+def feature_freeze_version(spec: str, collection_name: str) -> str:  # noqa: C901
```

Comment:
The code complexity linting errors could probably be fixed by making this a class and splitting some of the code into separate methods, but meh.

Reply:
What do you think of af943890611ddf5986839ef8c094b8788a660189?

Reply:
I like it, but the methods could use docstrings.